javabridge/locate.py

```diff
@@ -15,14 +15,39 @@
 import logging
 import subprocess
 
+
+# Note: if a matching gcc is available from the shell on Windows, its
+#       probably safe to assume the user is in an MINGW or MSYS or Cygwin
+#       environment, in which case he/she wants to compile with gcc for
+#       Windows, in which case the correct compiler flags will be triggered
+#       by is_mingw. This method is not great, improve it if you know a
+#       better way to discriminate between compilers on Windows.
+def is_mingw():
+    if (os.name=="nt"):
+        mingw32 = ""
```

Comment:
I've just updated the PR to include such a change. Now it should be rather conservative on guessing mingw.